src/Common.UnitTests/Handlers/NewEnvironmentHandlerTests.cs

```diff
@@ -210,4 +211,53 @@ await newEnvironmentHandler.Run(name, parameters, environmentType, siteName, lic
                 x => x.Count() == 1 && x[0] == cPSoftwareLicense.Id
             )), Times.Once);
     }
+    
+    [Theory]
+    [InlineData(UniversalState.Created, DeploymentStatus.DeploymentSucceeded)]
+    [InlineData(UniversalState.Active, DeploymentStatus.NotDeployed)]
+    [InlineData(UniversalState.Created, DeploymentStatus.NotDeployed)]
+    public async Task Run_WhenUniversalStateAndStatusRequiresCreation_CreatesOrNotCustomerEnvironment(UniversalState universalState, DeploymentStatus status)
+    {
+        // Arrange
+        using var mock = AutoMock.GetLoose();
+
+        var customerEnvironment = new CustomerEnvironment
+        {
+            Name = name,
+            UniversalState = universalState,
+            Status = status
+        };
+
+        bool shouldCallCreate = universalState != UniversalState.Created || status != DeploymentStatus.NotDeployed;
```

Comment:
I don't think having logic in our unit tests regarding how our code should be tested is a good practice. For this I would create 2 methods a `Run_WhenUniversalStateAndStatusRequiresCreation_CreateCustomerEnvironment` and a `Run_WhenUniversalStateAndStatusRequiresCreation_DoNotCreateCustomerEnvironment` with each method testing a particular scenario. Because this condition `shouldCallCreate` is exactly the change made in the code base, so we are not exactly testing the new logic we are just replicating it in our unit test.
Another option would be to just pass the Times.Once()... as a 3rd argument to the test method and just eliminate the `shouldCallCreate` login.

Reply:
In any case, I believe you have a point, and this should be more explicit! I've updated the test to include a third explicit argument.